File: backend/app/api/v1/static_analysis.py
from fastapi import APIRouter, Depends
from app.core.dependencies import get_current_user
from app.core.responses import StandardJSONResponse
from app.models.user import User

router = APIRouter(prefix="/static-analysis", tags=["static-analysis-configurations"])


@router.get("/config", response_class=StandardJSONResponse)
async def get_static_analysis_config(current_user: User = Depends(get_current_user)):
    """Retrieves enabled rules and configuration options for pylint, bandit, and radon checkers."""
    return {
        "checkers": {
            "pylint": {
                "enabled": True,
                "disabled_warnings": ["missing-docstring", "invalid-name"],
                "max_line_length": 120,
            },
            "bandit": {"enabled": True, "confidence_level": "medium", "severity_level": "medium"},
            "radon": {"enabled": True, "min_maintainability_index": "B", "max_cyclomatic_complexity": 10},
        }
    }

File: frontend/src/services/staticAnalysis.ts
import api from './api.ts';
import { ApiResponse } from '../types';

export interface CheckerConfig {
  enabled: boolean;
  [key: string]: any;
}

export interface StaticAnalysisConfig {
  checkers: {
    pylint: CheckerConfig;
    bandit: CheckerConfig;
    radon: CheckerConfig;
  };
}

export const getStaticAnalysisConfig = async (): Promise<ApiResponse<StaticAnalysisConfig>> => {
  const response = await api.get('/static-analysis/config');
  return response.data;
};
